pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -0,0 +1,205 @@
+`
+}
+
+func (c *DriftStatefilesCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "gcsBucket",
+		Target:  &c.flagGCSBucket,
+		Example: "my-gcs-bucket",
+		Usage:   "The gcs bucket to compare against local backend configurations. If none is provided then we will parse it from the backend config.",
+	})
+
+	return set
+}
+
+func (c *DriftStatefilesCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	parsedArgs := f.Args()
+
+	if len(parsedArgs) != 1 {
+		return flag.ErrHelp
+	}
+
+	dirAbs, err := util.PathEvalAbs(parsedArgs[0])
+	if err != nil {
+		return fmt.Errorf("failed to absolute path for directory: %w", err)
+	}
+	c.directory = dirAbs
+
+	c.gitClient = git.NewGitClient(c.directory)
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithRetryInitialDelay(c.RetryFlags.FlagRetryInitialDelay),
+		github.WithRetryMaxAttempts(c.RetryFlags.FlagRetryMaxAttempts),
+		github.WithRetryMaxDelay(c.RetryFlags.FlagRetryMaxDelay),
+	)
+	c.terraformParser, err = parser.NewTerraformParser(ctx, "")
+	if err != nil {
+		return fmt.Errorf("failed to create terraform parser: %w", err)
+	}
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian init process.
+func (c *DriftStatefilesCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx).
+		Named("drift.statefiles").
+		With("github_owner", c.GitHubFlags.FlagGitHubOwner).
+		With("github_repo", c.GitHubFlags.FlagGitHubOwner)
+
+	logger.Debug("starting Guardian drift statefiles")
+	logger.Debug("finding entrypoint directories")
+
+	entrypoints, err := terraform.GetEntrypointDirectories(c.directory)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+	entrypointBackendFiles := make([]string, 0, len(entrypoints))
+	for _, e := range entrypoints {
+		entrypointBackendFiles = append(entrypointBackendFiles, e.BackendFile)
+	}
+	logger.Debugw("terraform entrypoint directories", "entrypoint_backend_files", entrypointBackendFiles)
+
+	expectedURIs := make([]string, 0, len(entrypointBackendFiles))
+	gcsBuckets := make([]string, 0, len(entrypointBackendFiles))
+	for _, f := range entrypointBackendFiles {
+		config, _, err := terraform.ParseBackendConfig(f)
+		if err != nil {
+			return fmt.Errorf("failed to parse Terraform backend config: %w", err)
+		}
+		if config.GCSBucket == nil || *config.GCSBucket == "" {
+			return fmt.Errorf("unsupported backend type for terraform config at %s - only gcs backends are supported", f)
+		}
+		gcsBuckets = append(gcsBuckets, *config.GCSBucket)
+		expectedURIs = append(expectedURIs, fmt.Sprintf("gs://%s/%s/default.tfstate", *config.GCSBucket, *config.Prefix))
+	}
```

Comment:
Nit: it might be a better UX to process all of the entrypointBackendFiles and return a multierror (errors.Join). That would give the user all their invalid backends at once, instead of one at a time.

Reply:
Done